File: benchmarks/tsb/bench_dataframe_nlargest_nsmallest.ts
import { nlargestDataFrame, nsmallestDataFrame } from "tsb";
import { DataFrame } from "tsb";
const N = 100_000;
const df = new DataFrame({
  a: Array.from({ length: N }, (_, i) => (i * 1337) % 100_007),
  b: Array.from({ length: N }, (_, i) => (i * 7919) % 100_003),
  c: Array.from({ length: N }, (_, i) => (i * 3571) % 99_991),
});
const WARMUP = 3;
const ITERS = 20;
for (let i = 0; i < WARMUP; i++) {
  nlargestDataFrame(df, 100, "a");
  nsmallestDataFrame(df, 100, "a");
}
const t0 = performance.now();
for (let i = 0; i < ITERS; i++) {
  nlargestDataFrame(df, 100, "a");
  nsmallestDataFrame(df, 100, "a");
}
const total = performance.now() - t0;
console.log(JSON.stringify({ function: "dataframe_nlargest_nsmallest", mean_ms: total / ITERS, iterations: ITERS, total_ms: total }));

File: benchmarks/pandas/bench_dataframe_nlargest_nsmallest.py
import pandas as pd, time, json
N = 100_000
df = pd.DataFrame({
    "a": [(i * 1337) % 100_007 for i in range(N)],
    "b": [(i * 7919) % 100_003 for i in range(N)],
    "c": [(i * 3571) % 99_991 for i in range(N)],
})
WARMUP = 3
ITERS = 20
for _ in range(WARMUP):
    df.nlargest(100, "a")
    df.nsmallest(100, "a")
t0 = time.perf_counter()
for _ in range(ITERS):
    df.nlargest(100, "a")
    df.nsmallest(100, "a")
total = (time.perf_counter() - t0) * 1000
print(json.dumps({"function": "dataframe_nlargest_nsmallest", "mean_ms": total / ITERS, "iterations": ITERS, "total_ms": total}))
